### PR TITLE
[3.4] Adds composer service

### PR DESCRIPTION
**Pull request by nunomaduro**
The PR addresses the a separation of concerns. Isolation composer logic from the installer command.

### CODE REVIEW

File: src/Contracts/Providers/Composer.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+interface Composer
+{
+    /**
+     * Pulls the provided package.
+     *
+     * @param  string $package
+     *
+     * @return $this
+     */
+    public function require(string $package): ComposerContract
```
Comment (GrahamCampbell) [inline src/Contracts/Providers/Composer.php]: Bad return type?

Comment (nunomaduro) [reply src/Contracts/Providers/Composer.php]: @GrahamCampbell Thanks for the return! Actually the previous PR was only a POC for @introwit.

Now the PR contain the real implementation. Please fell free to perform the review! 😊

You are more than welcome!

File: src/Contracts/Providers/Composer.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * (c) Nuno Maduro <enunomaduro@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace NunoMaduro\ZeroFramework\Contracts\Providers;
+
+/**
+ * The is the Zero Framework composer contract.
```
Comment (svenluijten) [inline src/Contracts/Providers/Composer.php]: Did you mean "*This* is the Zero Framework composer contract."? 😉

Comment (nunomaduro) [reply src/Contracts/Providers/Composer.php]: @svenluijten Yeah. Nice catch 👍

**Pull request merged**